lib/Crunch.ML

```diff
@@ -117,21 +117,25 @@ val crunch_ignoreP =
     Outer_Syntax.local_theory
          @{command_keyword "crunch_ignore"}
         "add to and delete from list of things that crunch should ignore in finding prerequisites"
-        ((Scan.optional (P.$$$ "(" |-- P.name --| P.$$$ ")") "" -- Scan.optional
+        ((Scan.optional (P.$$$ "(" |-- P.list1 P.name --| P.$$$ ")") [""] -- Scan.optional
           (P.$$$ "(" |-- read_sections [add_sect, del_sect] --| P.$$$ ")")
-          []
-        )
-        >> (fn (crunch_instance, wpigs) => fn lthy =>
+          [])
+         >> (fn (crunch_instances, wpigs) => fn lthy =>
                let fun const_name const = dest_Const (read_const lthy const) |> #1;
                    val add = wpigs |> filter (fn (s,_) => s = #1 add_sect) |> map #2 |> constants
                                    |> map const_name;
                    val del = wpigs |> filter (fn (s,_) => s = #1 del_sect) |> map #2 |> constants
                                    |> map const_name;
-                   val crunch_ignore_add_del = (case get_crunch_instance crunch_instance (Proof_Context.theory_of lthy) of
-                     NONE => error ("Crunch has not been defined for " ^ crunch_instance)
-                   | SOME x => snd x);
+                   fun crunch_ignore_add_del inst =
+                     (case get_crunch_instance inst (Proof_Context.theory_of lthy) of
+                        NONE => error ("Crunch has not been defined for " ^ inst)
+                      | SOME x => snd x);
+                   val crunch_ignore_add_dels =
+                     map (fn inst => crunch_ignore_add_del inst add del) crunch_instances
+                   val crunch_ignore_add_del' =
```

Comment:
I thought for a moment that `crunch_ignore_add_del'` isn't all that informative, but reading over the flow of the function it's ok.

Reply:
Hmm, rereading it again, I think I prefer `crunch_ignore_add_dels'`. That's not really any more informative, but makes more sense to me.